examples/with_winit/src/stats.rs

```diff
@@ -0,0 +1,169 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Also licensed under MIT license, at your choice.
+
+use scenes::SimpleText;
+use std::{collections::VecDeque, time::Instant};
+use vello::{
+    kurbo::{Affine, Rect},
+    peniko::{Brush, Color, Fill},
+    SceneBuilder,
+};
+
+const SLIDING_WINDOW_SIZE: usize = 100;
+
+#[derive(Debug)]
+pub struct Snapshot {
+    pub fps: f64,
+    pub frame_time_ms: f64,
+    pub frame_time_min_ms: f64,
+    pub frame_time_max_ms: f64,
+}
+
+impl Snapshot {
+    pub fn draw_layer(
+        &self,
+        sb: &mut SceneBuilder,
+        text: &mut SimpleText,
+        viewport_width: f64,
+        viewport_height: f64,
+    ) {
+        let width = (viewport_width * 0.4).max(200.).min(400.);
+        let height = width * 0.3;
+        let x_offset = viewport_width - width;
+        let y_offset = viewport_height - height;
+        let offset = Affine::translate((x_offset, y_offset));
+        let text_height = height * 0.2;
+        let left_margin = width * 0.03;
+        let text_size = (text_height * 0.9) as f32;
+        sb.fill(
+            Fill::NonZero,
+            offset,
+            &Brush::Solid(Color::rgba8(0, 0, 0, 200)),
+            None,
+            &Rect::new(0., 0., width, height),
+        );
+        text.add(
+            sb,
+            None,
+            text_size,
+            Some(&Brush::Solid(Color::WHITE)),
+            offset * Affine::translate((left_margin, text_height)),
+            &format!("Frame Time: {:.2} ms", self.frame_time_ms),
+        );
+        text.add(
+            sb,
+            None,
+            text_size,
+            Some(&Brush::Solid(Color::WHITE)),
+            offset * Affine::translate((left_margin, 2. * text_height)),
+            &format!("Frame Time (min): {:.2} ms", self.frame_time_min_ms),
+        );
+        text.add(
+            sb,
+            None,
+            text_size,
+            Some(&Brush::Solid(Color::WHITE)),
+            offset * Affine::translate((left_margin, 3. * text_height)),
+            &format!("Frame Time (max): {:.2} ms", self.frame_time_max_ms),
+        );
+        text.add(
+            sb,
+            None,
+            text_size,
+            Some(&Brush::Solid(Color::WHITE)),
+            offset * Affine::translate((left_margin, 4. * text_height)),
+            &format!("Resolution: {viewport_width}x{viewport_height}"),
+        );
+        text.add(
+            sb,
+            None,
+            text_size,
+            Some(&Brush::Solid(Color::WHITE)),
+            offset * Affine::translate((width * 0.67, text_height)),
+            &format!("FPS: {:.2}", self.fps),
+        );
+    }
```

Comment:
Could we also render a frame time bar graph, showing how stable we are, maybe with some neat colour coding?

Reply:
This is absolutely on my TODO list. I'll spend some time this week to incorporate this into this PR (should be easy to add since the code to track samples for the bar graph is already there).

Reply:
Done. For now all bars are drawn as green but I can look into some color coding next.